### PR TITLE
Test Coverage Reports for E2E tests

### PR DESCRIPTION
**Pull request by dereknola**
#### Proposed Changes ####
- Move coverage logic from the server into the startup for both agent and server
- Pin go version used for E2E `go test` to latest stable release
- Add test coverage reporting to validate, secrets-encrypt and startup E2E tests

#### Types of Changes ####
E2E

#### Verification ####
Drone CI passes, see increase in coverage report https://app.codecov.io/gh/k3s-io/k3s/pull/7526

#### Testing ####
Its all testing

#### Linked Issues ####
N/A

#### User-Facing Change ####

```release-note

```

### CODE REVIEW

File: tests/e2e/vagrantdefaults.rb [hunk 1]
```diff
@@ -34,6 +34,17 @@ def getInstallType(vm, release_version, branch)
   end
 end
 
+def addCoverageDir(vm, role)
+  service = role.include?("agent") ? "k3s-agent" : "k3s" 
+    script = <<~SHELL
+      mkdir -p /tmp/k3scov
+      mkdir -p /etc/systemd/system/#{service}.service.d
```
Comment (brandond) [inline tests/e2e/vagrantdefaults.rb:41]: You shouldn't need to create an override file, just put the env variable into one of the supported env files:
https://github.com/k3s-io/k3s/blob/2b24c9917cdfec92439ac68dd6706fffe20195a5/install.sh#L846-L848

File: .drone.yml [hunk 2]
```diff
@@ -765,6 +768,25 @@ steps:
   - name: cache
     path: /tmp/artifacts
 
+- name: upload to codecov
+  image: robertstettner/drone-codecov
```
Comment (brandond) [inline .drone.yml:772]: I love images that just are just from some dude

**Review by brandond — APPROVED** [review]

File: .drone.yml (hunk 2 above)
Comment (brandond) [reply .drone.yml:774]: the repo hasn't been touched in 6 years, is the image also that old!?

Comment (brandond) [reply .drone.yml:774]: It looks like it's also just a simple wrapper around the deprecated bash uploader; I wonder if we should just toss together a proper uploader that uses the new one from https://docs.codecov.com/docs/codecov-uploader

Comment (dereknola) [reply .drone.yml:774]: I'll open a future PR to move to directly uploading via their bash tool

Comment (brandond) [reply .drone.yml:774]: no, bash is bad, we want to use the new non-bash one.

**Review by osodracnai — APPROVED** [review]

**Pull request merged**